pkg/model/resources/nodeup.go

```diff
@@ -64,9 +64,11 @@ function ensure-install-dir() {
 
 # Retry a download until we get it. args: name, sha, urls
 download-or-bust() {
+  echo "== Downloading $1 with hash $2 from $3 =="
   local -r file="$1"
   local -r hash="$2"
-  local -r urls=( $(split-commas "$3") )
+  local -a urls
+  mapfile -t urls < <(split-commas "$3")
```

Comment:
It looks like mapfile is a bash-ism, but I think we're already relying on bash-specific features.

Reply:
See: https://www.shellcheck.net/wiki/SC2207